### PR TITLE
Bring down latest dependencies when publishing

### PR DESCRIPTION
**Pull request by kipz**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,7 +11,8 @@ deploy:
     secure: E0PfPJ6MlbR6YsVsg4zi0XGxv4sgOk9IZgEcRK6cdl/BETRKCqjpB3ZrJhR4khB43sbw4ojKt3kmjDN5LOANza1xe5mJkdYyiodcaivAIMk51YRgdVcKr78cKvJegIZoBQb6jsKw4PDdnECsdpOGaAOWD0zdVbcNx7AofE8NqApOA00Y+IDEtE8Hvl/kEqgrFpO0rGvCLMTWCpKbsycDzHrt/0lpuOSvRz6NahS90ypz2udrF/zPwMZ3SdMMUosOD0yaWZgA7VXVWHfTyVdBkHFYswwtlWjsoK0XjoP+DIIuSCPGE0+OIVESSCyTw7rmuyMsnfCSxsvdkLLiPn0uEmhDKx2VvtqKLM2c4mblMIyCEf4t1nIOItqWlJiAS01VvYWdMQpkbXbJflpptMsPdbkcuJnrHgwlo8I6+hZ0Q3psewB4UBaGn55K0yXuvCtabq5aaMt5v5jwzEK5ZYUgtRNfdsO4jNeRPrSPoSyMKH/fBS7lH8X5w+eYplaMNuguHs+2EWopxIxCrV81Zn7DZ92JZshxRYxZHYgmULPOCtFqRREna5XzGYoH20iR4HSzpqnLbmXQCvi03tgci9MDc6zX4ONU1k9o84qg6r1xSwCRPIptvTpqum20rpAtjdsh1jtjSMwZcOfvrswmX21bQOme7HjR7sjM8sv9n5sbfgg=
   on:
     tags: true
-    condition: $TRAVIS_TAG =~ ^[0-9]+\.[0-9]+\.[0-9]+$
```
Comment (ddgenome) [inline .travis.yml:12]: We definitely do not want to remove this. Removing this means a GitHub release will be created on every build.

File: npm-publish.bash [hunk 2]
```diff
@@ -14,36 +14,52 @@ function err() {
...
         return 10
     fi
 
-    local package="package.json"
-    local tmp_package="$package.tmp"
     if ! jq --arg tag "$TRAVIS_TAG" '.version = $tag' "$package" > "$tmp_package"; then
         err "failed to set version in $package"
         return 1
     fi
     cp "$tmp_package" "$package"
```
Comment (ddgenome) [inline npm-publish.bash:30]: The exit value of this statement should be checked.

File: npm-publish.bash [hunk 3]
```diff
@@ -14,36 +14,52 @@ function err() {
...
-    rm -f "$target/.gitignore"
+        if ! ( cd "$target" && npm publish --access=public ); then
+            err "failed to publish node module"
+            cat "$target/npm-debug.log"
+            return 1
+        fi
+    else [[ $TRAVIS_BRANCH == "master" ]]; then
+        msg "Creating local .npmrc from details generated in Artifactory..."
+        if [[ $ATOMIST_REPO_TOKEN  && $ATOMIST_REPO_USER ]]; then
+            if ! ( curl -u"${ATOMIST_REPO_USER}":"${ATOMIST_REPO_TOKEN}" "https://atomist.jfrog.io/atomist/api/npm/auth" >  "$HOME/.npmrc"); then
```
Comment (ddgenome) [inline npm-publish.bash:49]: The parentheses should not be necessary.

The quiet option should be used with curl and stderr should be redirected to `/dev/null` so the user name and password do not leak into the logs.

File: update-deps.bash [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+function err() {
+    msg "$*" 1>&2
+}
+
+function main() {
+    msg "branch is ${TRAVIS_BRANCH}"
+
+    if [[ $TRAVIS_TAG =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+        msg "Updating @atomist modules to latest released versions..."
+        if ! npm install @atomist/rug@latest -S; then
```
Comment (ddgenome) [inline update-deps.bash:21]: Is there a point in using the `-S` option if we are not committing the changes to `package.json`?

**Review by ddgenome — CHANGES_REQUESTED** [review]
We definitely need to restore the release condition.

File: npm-publish.bash (hunk 2 above)
Comment (kipz) [reply npm-publish.bash:30]: I agree, but this was a copy/paste.

File: update-deps.bash (hunk 4 above)
Comment (kipz) [reply update-deps.bash:21]: The point is that they end up in the package.json in the node module, but actually I think an editor should do this, and we should _not_ be doing auto-updates on releases. I'll revert that bit.

Comment (ddgenome) [reply update-deps.bash:21]: Makes sense, auto-update on snapshot builds but not releases.

**Comment by kipz** [comment]
Over to you.

**Pull request closed without merging**